core/src/main/java/org/apache/iceberg/V1Metadata.java

```diff
@@ -330,32 +333,44 @@ public ManifestEntry<DataFile> copyWithoutStats() {
     }
   }
 
-  static class IndexedDataFile implements DataFile, IndexedRecord {
+  static class DataFileWrapper implements DataFile, StructLike {
     private static final long DEFAULT_BLOCK_SIZE = 64 * 1024 * 1024;
 
-    private final org.apache.avro.Schema avroSchema;
-    private final IndexedStructLike partitionWrapper;
+    private final int size;
     private DataFile wrapped = null;
 
-    IndexedDataFile(org.apache.avro.Schema avroSchema) {
-      this.avroSchema = avroSchema;
-      this.partitionWrapper = new IndexedStructLike(avroSchema.getField("partition").schema());
```

Comment:
There is also no need for a wrapper to adapt `PartitionData` to `IndexedRecord` because it is already `StructLike`.

Reply:
Big fan of this change